### ISSUE TITLE
Multiple issues when application path contains non-ascii characters

### ISSUE THREAD

**Issue by tyrylu** [manuelcortez/TWBlue#88]
Hello.
During a test of the portable version (latest stable), i found out that there are some issues when the app path contains non-ascii chars. This can happen for example in the Czech republic, where the user name (and thus the downloads folder) quite often contains something beyond ascii.
I was able to fix the issues enough to start the application, but the fixes are quite hackish, at least sometimes, but see for yourself:
1. Libloader is not capable of handling unicode paths, so when loading the libraries, with keys, it is necessary to encode the paths (see 4. for reason why the data type changed).
2. Someone is passing a str to Pybass_LoadPlugin, when it expects unicode strings, i was not able to determine who (but likely sound_lib/the infrastructure around it).
3. Win32com expects an encoded path in the **gen_path** variable. Or, better, it expects it first, but then it fails when an attempt is made to create a COM object.
4. In your paths.py, it's necessary to decode at least the app_path, because further attempts to call os.path.join fail otherwise. Of course, it then causes issue number 1., but there's probably nothing you can do about that than account for it.

I know, at least 3. is a bug in accessible_output2/libloader/may be win32com, so i am not sure what's your policy of fixing those.
And despite all those, closing the app when the cache should be saved does not work.
Sure, it may be easyer to add a warning to the readme, it's up to you, but if you decide to fix it, perhaps i could help.

**Comment by manuelcortez** [comment]
Thanks for your report. I will take a look. Basically accessible_output2 is using win32com for making some screen readers to speak, that's why it could cause that issue. I will investigate this problem, and create a path with non-ascii characters for testing this problem. About the bug with closing the application while the cache is enabled, it shouldn't happen. I think it's much better to try to fix this issue instead putting a warning message.

**Comment by tyrylu** [comment]
I agree with that. I know, the issues in the third party libraries are always harder to solve, not including the community openness (or lack thereof) of accesible_output2 and friends.
